src/store/todo.store.js

```diff
@@ -0,0 +1,68 @@
+import { setTodos } from '../service/todo.service.js';
+import { changeFilterList } from '../ui/todo/todo.dom.js';
+export const addTodo = function (value) {
+  const timestamp = Date.now();
+  const todo = {
+    key: timestamp,
+    value,
+    complete: false,
+  };
+  this.todos.unshift(todo);
+
+  setTodos(this.id, this.todos);
+};
+
+export const changeCompleteTodo = function (targetKey) {
+  let targetTodo = [];
+
+  const filteredTodos = this.todos.filter((todo) => {
+    if (todo.key === targetKey) {
+      targetTodo = todo;
+      targetTodo.complete = !targetTodo.complete;
+    } else {
+      return todo;
+    }
+  });
```

Comment:
p5. filter 돌릴 때 undefined를 리턴하기보다 다음과 같이 처리하면 좀 더 조건을 명확히 이해할 수 있을 것 같아 보여요! 추가로 targetTodo에서 객체를 할당해주는 것 같은데, 혹시 배열로 초기화하신 이유가 따로 있나요??

```suggestion
  let targetTodo = null;

  const filteredTodos = this.todos.filter((todo) => {
    if (todo.key === targetKey) {
      targetTodo = todo;
      targetTodo.complete = !targetTodo.complete;
    }

    return todo.key !== targetKey;
  });
```